src/extension.ts

```diff
@@ -9,11 +9,12 @@ async function pickConfig(manager: Manager, activeOrNot?: boolean) {
   if (activeOrNot === false) {
     names = others.filter(c => !names.find(cc => cc.name === c.name));
   } else if (activeOrNot === undefined) {
-    others.forEach(n => names.indexOf(n) === -1 && names.push(n));
+    names = others;
   }
   const options: vscode.QuickPickItem[] = names.map(config => ({
-    label: config.label || config.name,
-    description: config.label && config.name,
+    label: config.label || config.username + '@' + config.host,
+    description: config.root,
+    detail: config.name
```

Comment:
I would rather keep the current visualization (description/label), and use detail to show the username, host, root, ... if known. Not all of these are immediately known _(PuTTY sessions, and later on, .ssh/config sessions)_